### PR TITLE
support --output-dir option for helm3 template

### PR DESCRIPTION
**Pull request by torstenwalter**
**What this PR does / why we need it**:

helm 2 has an option to write the executed templates to files in output-dir instead of stdout.
This PR adds the same to helm3.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by jdolitsky — APPROVED** [review]
LGTM

File: pkg/action/install.go [hunk 1]
```diff
@@ -363,12 +366,56 @@ func (c *Configuration) renderResources(ch *chart.Chart, values chartutil.Values
 
 	// Aggregate all valid manifests into one big doc.
 	for _, m := range manifests {
-		fmt.Fprintf(b, "---\n# Source: %s\n%s\n", m.Name, m.Content)
+		if outputDir == "" {
```
Comment (hickeyma) [inline pkg/action/install.go]: Should it be checked here for a blank template? This was similar to how it was checked in v2:
```console
if whitespaceRegex.MatchString(m.Content) {
        continue
}
```

File: pkg/action/install.go [hunk 2]
```diff
@@ -363,12 +366,56 @@ func (c *Configuration) renderResources(ch *chart.Chart, values chartutil.Values
...
+			fmt.Fprintf(b, "---\n# Source: %s\n%s\n", m.Name, m.Content)
+		} else {
+			err = writeToFile("build", m.Name, m.Content)
```
Comment (hickeyma) [inline pkg/action/install.go]: :s/"build"/outputDir

**Review by hickeyma — CHANGES_REQUESTED** [review]
@torstenwalter Thanks for submitting the PR. I have added some comments for change.
I think it would also be helpful if you could add a test case.

**Comment by torstenwalter** [comment]
@hickeyma Thanks for your feedback. The suggested changes have been incorporated.

**Comment by hickeyma** [comment]
@torstenwalter Would it be possible to add a unit test as mentioned in https://github.com/helm/helm/pull/5762#pullrequestreview-242671847 ?

**Comment by torstenwalter** [comment]
@hickeyma Do have anything I can use as example? Haven't found any unit test in helm 2 for this.

**Comment by torstenwalter** [comment]
@hickeyma While writing the test I fugured out that the check for empty manifests had no effect. They are never empty.

That's already implemented in
https://github.com/helm/helm/blob/dev-v3/pkg/releaseutil/manifest.go#L49-L51 and in https://github.com/helm/helm/blob/dev-v3/pkg/releaseutil/manifest_sorter.go#L88-L90

**Comment by torstenwalter** [comment]
If you want I can also rebase and squash my changes on `dev-v3` branch.

**Review by hickeyma — APPROVED** [review]
LGTM, thanks for all your effort.

**Pull request merged**